clang/utils/TableGen/NeonEmitter.cpp

```diff
@@ -2580,6 +2580,8 @@ void NeonEmitter::runVectorTypes(raw_ostream &OS) {
   OS << "typedef __fp16 float16_t;\n";
 
   OS << "#if defined(__aarch64__) || defined(__arm64ec__)\n";
+  OS << "typedef __MFloat8x8_t mfloat8x8_t;\n";
+  OS << "typedef __MFloat8x16_t mfloat8x16_t;\n";
```

Comment:
Is there a reason for the new types to be guarded?  When looking at `arm_vector_types.h` the guards are only used for `float64` based types, with the other vector types unguarded.

Also, and this might necessitate reintroducing some of the code you were asked to remove, I'd rather `emitNeonTypeDefs` have the capability to emit the typedefs because that'll verify we have the necessary `TypeSpec` plumbing required to add the builtins.

Reply:
That make sense. I was keen to remove the `TypeSpec` stuff as it was not used at all, but these typedefs should probably be emitted by `emitNeonTypeDefs` as they are.. well, still Neon types after all.

Reply:
So if we dont want to protect that to 64 bits architecture, then I have to change Sema and ASTContext to introduce the neon type for other architectures that are not 64 bits. Otherwise they fail, because the typedef and the builtin are  only introduced/added when Target.hasAArch64SVETypes.

Reply:
Oh, I hadn't realised this was a shared header with `arm` as well as `aarch64`.  Fair enough, I retract my comment.

Reply:
Is it still possible to make use of `emitNeonTypeDefs`? I can see that also emits the inclusion guards when necessary.

Reply:
Based on an offline conversation it has been agreed to wait for the scalar type to land before trying to extend `emitNeonTypeDefs` and thus this PR can continue as is..